src/components/multi-action-button/multi-action-button.test.tsx

```diff
@@ -0,0 +1,157 @@
+      <Button>First</Button>
+    </MultiActionButton>
+  );
+
+  expect(screen.getByTestId("bar")).toHaveAttribute("data-element", "foo");
+});
+
+test("should render with provided 'id'", () => {
+  render(
+    <MultiActionButton text="Main Button" id="test">
+      <Button>First</Button>
+    </MultiActionButton>
+  );
+
+  expect(screen.getByRole("button", { name: "Main Button" })).toHaveAttribute(
+    "id",
+    "test"
+  );
+});
+
+test("should call 'onClick' when the main button is clicked", async () => {
+  const onClick = jest.fn();
+  const user = userEvent.setup();
+  render(
+    <MultiActionButton text="Main Button" onClick={onClick}>
+      <Button>First</Button>
+    </MultiActionButton>
+  );
+
+  await user.click(screen.getByRole("button", { name: "Main Button" }));
+
+  expect(onClick).toHaveBeenCalledTimes(1);
+});
+
+test("should open additional buttons when the main button is clicked", async () => {
+  const user = userEvent.setup();
+  render(
+    <MultiActionButton text="Main Button">
+      <Button>First</Button>
+    </MultiActionButton>
+  );
+
+  await user.click(screen.getByRole("button", { name: "Main Button" }));
+
+  expect(await screen.findByRole("button", { name: "First" })).toBeVisible();
+});
+
+test("should close additional buttons when a child button is clicked", async () => {
+  const user = userEvent.setup();
+  render(
+    <MultiActionButton text="Main Button">
+      <Button>First</Button>
+    </MultiActionButton>
+  );
+
+  await user.click(screen.getByRole("button", { name: "Main Button" }));
+  await user.click(screen.getByRole("button", { name: "First" }));
+
+  expect(
+    screen.queryByRole("button", { name: "First" })
+  ).not.toBeInTheDocument();
+});
+
+test("should close additional buttons when a click occurs outside the component", async () => {
+  const user = userEvent.setup();
+  render(
+    <MultiActionButton text="Main Button">
+      <Button>First</Button>
+    </MultiActionButton>
+  );
+
+  await user.click(screen.getByRole("button", { name: "Main Button" }));
+  await user.click(document.body);
+
+  expect(
+    screen.queryByRole("button", { name: "First" })
+  ).not.toBeInTheDocument();
+});
+
+test("should render when children are non-Carbon Button children", async () => {
+  const user = userEvent.setup();
+  render(
+    <MultiActionButton text="Main Button">
+      <span>First</span>
+    </MultiActionButton>
+  );
+
+  await user.click(screen.getByRole("button", { name: "Main Button" }));
+
+  expect(screen.getByText("First")).toBeVisible();
+});
+
+test("should render main button as disabled when 'disabled' prop is true", () => {
+  render(
+    <MultiActionButton text="Main Button" disabled>
+      <Button>First</Button>
+    </MultiActionButton>
+  );
+
+  expect(screen.getByRole("button", { name: "Main Button" })).toBeDisabled();
+});
+
+test("should render with expected styles when 'width' prop is set", () => {
```

Comment:
**question**: is this necessary for coverage?

Reply:
yeah it is 😞